### PR TITLE
Support `lastEvaluation` and `evaluationTime` in `/api/v1/rules` endpoints and make order of groups stable.

### PR DESCRIPTION
**Pull request by Wing924**
**What this PR does**:

Support `lastEvaluation` and `evaluationTime` in `/api/v1/rules` endpoints and make order of groups stable.

**Which issue(s) this PR fixes**:

Without `lastEvaluation` and `evaluationTime`, `/api/prom/new/rules` show NaN in prometheus UI.

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,7 +66,7 @@
 * [BUGFIX] The gauge `cortex_overrides_last_reload_successful` is now only exported by components that use a `RuntimeConfigManager`. Previously, for components that do not initialize a `RuntimeConfigManager` (such as the compactor) the gauge was initialized with 0 (indicating error state) and then never updated, resulting in a false-negative permanent error state. #2092
 * [BUGFIX] Fixed WAL metric names, added the `cortex_` prefix.
 * [BUGFIX] Restored histogram `cortex_configs_request_duration_seconds` #2138
-* [BUGFIX] Fix wrong syntax for `url` in config-file-reference. #2148
+* [ENHANCEMENT] Fix wrong syntax for `url` in config-file-reference. #2148
```
Comment (jtlisi) [inline CHANGELOG.md]: This doesn't seem related to this PR?

Comment (Wing924) [reply CHANGELOG.md]: I've fixed it.

File: pkg/ruler/ruler.go [hunk 2]
```diff
@@ -560,7 +566,7 @@ func (r *Ruler) getShardedRules(ctx context.Context, userID string) ([]*rules.Ru
 		return nil, fmt.Errorf("unable to inject user ID into grpc request, %v", err)
 	}
 
-	rgs := []*rules.RuleGroupDesc{}
+	rgs := make([]*rules.RuleGroupDesc, 0, len(rulers.Ingesters))
```
Comment (jtlisi) [inline pkg/ruler/ruler.go]: I don't think you need this change. The number of Rulers in the ring does not correlate to the number of rule groups being evaluated.

Comment (Wing924) [reply pkg/ruler/ruler.go]: > The number of Rulers in the ring does not correlate to the number of rule groups being evaluated.

Yes, but it should be no more than `len(rulers.Ingesters)`. This fix is for avoid realloc slice.
if the slice is empty (before my fix), the len and cap is `0`. After each append, the len increase 1 but cap will be 0, 2, 2, 4, 4..., which make malloc many times.

Comment (gouthamve) [reply pkg/ruler/ruler.go]: Can you add a comment for the same? I'll merge it after!

Comment (Wing924) [reply pkg/ruler/ruler.go]: I've added the comment.
